apps/web-client/src/lib/__testData__/debugHandlers.ts

```diff
@@ -0,0 +1,226 @@
+import { upsertBook } from "$lib/db/cr-sqlite/books";
+import { upsertCustomer } from "$lib/db/cr-sqlite/customers";
+import { createReconciliationOrder, upsertReconciliationOrderLines } from "$lib/db/cr-sqlite/order-reconciliation";
+import { createSupplierOrder } from "$lib/db/cr-sqlite/suppliers";
+import type { Customer, DB, PossibleSupplierOrderLine, Supplier } from "$lib/db/cr-sqlite/types";
+import type { BookData } from "@librocco/shared";
+
+/**
+* Upserts multiple customers into the database in a single transaction.
+* Each customer must have an id and displayId.
+* Updates the customers' lastModified timestamp automatically.
+*
+* @param {DB} db - Database connection
+* @param {Array<Omit<Customer, "updatedAt">>} customers - Array of customer
+data objects
+* @returns {Promise<{success: boolean, inserted: number, updated: number,
+errors: Array<{id: string, error: string}>}>}
+* @throws {Error} If transaction fails
+*/
+export async function upsertCustomers(db: DB, customers: Array<Omit<Customer, "updatedAt">>): Promise<void> {
+	if (!customers || !Array.isArray(customers) || customers.length === 0) {
+		return;
+	}
+
+	const errors: Array<{ id: number; error: string }> = [];
+
+	return await db.tx(async (txDb) => {
+		for (const customer of customers) {
+			// Validate required fields
+			if (!customer.id) {
+				errors.push({ id: parseInt(customer.displayId), error: "Customer must have an id" });
+				continue;
+			}
+
+			if (!customer.displayId) {
+				errors.push({ id: customer.id, error: "Customer must have a displayId" });
+				continue;
+			}
+
+			await upsertCustomer(txDb, customer);
+		}
+	});
+}
+
+/**
+ * Creates or updates multiple book records in a single transaction.
+ * Uses ISBN as the unique identifier for upsert operations.
+ * All fields except ISBN are optional and will only be updated if provided.
+ *
+ * @param {DB} db - Database connection
+ * @param {BookData[]} books - Array of book metadata objects
+ * @throws {Error} If any book is missing an ISBN
+ * @returns {Promise<void>}
+ */
+export async function upsertBooks(db: DB, books: BookData[]): Promise<void> {
+	if (!books.length) {
+		return;
+	}
```

Comment:
I think this should be done directly in debug page:
- this function is defined here, along with the data
- it is used in only one place, with data defined right next to it
- uses `db.tx` lightly, which could cause problems if the underlaying function (e.g. `upsertBook`) uses `db.tx` as  well -- this isn't the case here, but there are other functions that do
- no need for checks/error aggregation: they will be performed (if legal) in the same way as any other update, in case of an error: it will be reported as any other update